packages/breadcrumb/src/react/index.js

```diff
@@ -1,24 +1,20 @@
 import * as glamor from 'glamor'
-import Button from '@pluralsight/ps-design-system-button/react'
-import Icon from '@pluralsight/ps-design-system-icon/react'
+import Button from '@pluralsight/ps-design-system-button/react.js'
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+import Icon from '@pluralsight/ps-design-system-icon/react.js'
 import PropTypes from 'prop-types'
 import React from 'react'
 
-import css from '../css'
+import css from '../css/index.js'
 
 const styles = {
   breadcrumb: _ => glamor.css(css['.psds-breadcrumb'])
 }
 
 const Breadcrumb = React.forwardRef((props, ref) => {
-  const breadcrumbProps = {
-    ...styles.breadcrumb(props),
-    ...(props.style ? { style: props.style } : null),
-    ...(props.className ? { className: props.className } : null)
-  }
-
+  const { onClick, ...rest } = props
   return (
-    <div {...breadcrumbProps}>
+    <div {...filterReactProps(rest)} {...styles.breadcrumb(props)}>
```

Comment:
line 17 and 19 in combo concern me.  

"all components shall receive any html-legal props" is an easy-to-say rule.  But I'm having a hard time doing it in a useful way.  

Before, we had className and style going to root.  Then all other props going to the child button.  

But what props will be sent?

I think className and style should always go to root.  Because that's the most flexible (still terrible) way to override styles on a component with multiple elements.

So maybe I should have put filter on the button??

Reply:
hmm...getting tricky. i think props going to the root sounds like the best approach. events would hopefully bubble up from the button. the ref i would probably say should be on the root as well...but that's going to break what we have currently so i would leave it for now. 

it would be interesting to see why users would need to add attrs to the button or get the reference - and if becomes something we want to support I think we could do it better with a compound component